deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -627,11 +627,23 @@ def merge_input_provider(index, item):
         input_inflation=input_inflation,
         input_average_record_size_bytes=input_average_record_size_bytes,
     )
-
     logger.info(
         f"partition-{params.source_partition_locator.partition_values},"
         f"compacted at: {params.last_stream_position_to_compact},"
     )
+    is_inplace_compacted: bool = (
+        params.source_partition_locator == params.destination_partition_locator
+    )
+    if (
+        is_inplace_compacted
+        and compacted_partition != rcf_source_partition_locator
+    ):
```

Comment:
Add context/document the condition and if it changes then describe how the second concurrent compaction may not succeed